lib/chat_api/notes/note.ex

```diff
@@ -0,0 +1,34 @@
+defmodule ChatApi.Notes.Note do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias ChatApi.Accounts.Account
+  alias ChatApi.Customers.Customer
+  alias ChatApi.Users.User
+
+  @primary_key {:id, :binary_id, autogenerate: true}
+  @foreign_key_type :binary_id
+  schema "notes" do
+    field :body, :string
+
+    belongs_to(:account, Account)
+    belongs_to(:customer, Customer)
+    belongs_to(:author, User, foreign_key: :author_id, references: :id, type: :integer)
+
+    timestamps()
+  end
+
+  @doc false
+  def create_changeset(note, attrs) do
```

Comment:
i think we can just use `changeset` below, this `create_changeset` shouldn't be necessary 👍 (reference: https://hexdocs.pm/ecto/Ecto.Changeset.html#validate_required/3)

> If a field is given to validate_required/3 but it has not been passed as parameter during cast/3 (i.e. it has not been changed), then validate_required/3 will check for its current value in the data. If the data contains a non-empty value for the field, then no error is added. **This allows developers to use validate_required/3 to perform partial updates. For example, on insert all fields would be required, because their default values on the data are all nil, but on update, if you don't want to change a field that has been previously set, you are not required to pass it as a parameter**, since validate_required/3 won't add an error for missing changes as long as the value in the data given to the changeset is not empty.

Reply:
Updated! My thought here was actually that allowing the entire changeset in update requests allows the client to change the Customer or the Account that the note displays on -- can add back if we're concerned about this!